python/tank_vendor/shotgun_deploy/descriptor.py

```diff
@@ -0,0 +1,296 @@
+    ###############################################################################################
+    # data accessors
+
+    def get_location(self):
+        """
+        Returns the location dict associated with this descriptor
+
+        :returns: Dictionary that can be used to construct the descriptor
+        """
+        return self._io_descriptor.get_location()
+
+    def get_uri(self):
+        """
+        Returns the location uri associated with this descriptor
+        The uri is a string based representation that is equivalent to the
+        location dictionary returned by the get_location() method.
+
+        :returns: Uri string that can be used to construct the descriptor
+        """
+        return self._io_descriptor.get_uri()
+
+    def copy(self, target_folder):
+        """
+        Copy the config descriptor into the specified target location
+
+        :param target_folder: Folder to copy the descriptor to
+        """
+        self._io_descriptor.copy(target_folder)
+
+    def get_display_name(self):
+        """
+        Returns the display name for this item.
+        If no display name has been defined, the system name will be returned.
+
+        :returns: Display name as string
+        """
+        meta = self._io_descriptor.get_manifest()
+        display_name = meta.get("display_name")
+        if display_name is None:
+            display_name = self.get_system_name()
+        return display_name
+
+    def is_developer(self):
+        """
+        Returns true if this item is intended for development purposes
+
+        :returns: True if this is a developer item
+        """
+        return self._io_descriptor.is_developer()
+
+    def get_description(self):
+        """
+        Returns a short description for the app.
+
+        :returns: Description as string
+        """
+        meta = self._io_descriptor.get_manifest()
+        desc = meta.get("description")
+        if desc is None:
+            desc = "No description available." 
+        return desc
+
+    def get_icon_256(self):
+        """
+        Returns the path to a 256px square png icon file for this app
+
+        :returns: Path to a 256px png icon file
+        """
+        app_icon = os.path.join(self._io_descriptor.get_path(), "icon_256.png")
+        if os.path.exists(app_icon):
+            return app_icon
+        else:
+            # return default
+            default_icon = os.path.abspath(os.path.join(
+                os.path.dirname(__file__),
+                "resources",
+                "default_bundle_256px.png"
+            ))
+            return default_icon
+
+    def get_support_url(self):
+        """
+        Returns a url that points to a support web page where you can get help
+        if you are stuck with this item.
+
+        :returns: Url as string or None if not defined
+        """
+        meta = self._io_descriptor.get_manifest()
+        support_url = meta.get("support_url")
+        if support_url is None:
```

Comment:
Not sure this is a good idea. Imagine a world where clients are hosting their configs on Github for other people to use. If they forget to fill in the field, we'll be the ones reported as doing support for it. I'd rather say let's fix the manifests of our stuff.